AppointmentsService/AppointmentsService/ServiceExtensions/ExceptionMiddleware.cs

```diff
@@ -0,0 +1,66 @@
+﻿using AppointmentsService.ServiceExtensions.Exceptions;
+using Microsoft.AspNetCore.Mvc;
+using System.Net;
+using System.Text.Json;
+
+namespace AppointmentsService.ServiceExtensions
+{
+    public class ExceptionMiddleware
+    {
+        private readonly RequestDelegate _next;
+
+        public ExceptionMiddleware(RequestDelegate next)
+        {
+            _next = next;
+        }
+
+        public async Task InvokeAsync(HttpContext context)
+        {
+            try
+            {
+                await _next(context);
+            }
+            catch (Exception ex)
+            {
+                if (ex is NotFoundException)
+                {
+                    context.Response.StatusCode = (int)HttpStatusCode.NotFound;
+                    context.Response.ContentType = "application/json";
+
+                    await WriteErrorMessage(ex, context);
+                }
+                else if (ex is BadRequestException)
+                {
+                    context.Response.StatusCode = (int)HttpStatusCode.BadRequest;
+                    context.Response.ContentType = "application/json";
+
+                    await WriteErrorMessage(ex, context);
+                }
+                else
+                {
+                    context.Response.StatusCode = (int)HttpStatusCode.InternalServerError;
+                    context.Response.ContentType = "application/json";
+
+                    await WriteErrorMessage(ex, context);
+                }
```

Comment:
Use a switch case where you will just set a` context.Response.StatusCode`
And after this just call  await `WriteErrorMessage(ex, context);`
also you can move `context.Response.ContentType = "application/json";`